File: app/src/main/java/unit/DisPlayUnit.java
package unit;

import android.content.Context;
import android.util.DisplayMetrics;
import android.view.WindowManager;

import com.example.anxiao.mytestapplication.HomeApplication;

/**
 * Created by anxiao on 2017/8/9.
 * 屏幕相关参数
 */

public class DisPlayUnit {

    public static DisplayMetrics devicesDisPlay() {
        DisplayMetrics metrics = new DisplayMetrics();
        Context context = HomeApplication.getInstance();
        WindowManager manager = (WindowManager) context.getSystemService(Context.WINDOW_SERVICE);
        manager.getDefaultDisplay().getMetrics(metrics);
        return metrics;
    }



}

File: app/src/main/java/unit/DisPlayUnit.kt
package unit

import android.content.Context
import android.util.DisplayMetrics
import android.view.WindowManager

import com.example.anxiao.app.HomeApplication

/**
 * Created by anxiao on 2017/8/9.
 * 屏幕相关参数
 */

object DisPlayUnit {

    fun devicesDisPlay(): DisplayMetrics {
        val metrics = DisplayMetrics()
        val context = HomeApplication.getInstance()
        val manager = context.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        manager.defaultDisplay.getMetrics(metrics)
        return metrics
    }

    fun dip2px(dpValue: Int): Int {
        val scale = HomeApplication.getInstance().resources.displayMetrics.density
        return (dpValue * scale + 0.5f).toInt()
    }


}
